File: Quizzical/src/main/java/com/asfour/modules/AppModule.java
package com.asfour.modules;

import android.app.Application;

import com.asfour.application.App;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Waqqas on 15/07/15.
 */
@Module
public class AppModule {

    private App mApp;

    public AppModule(App app){
        mApp = app;
    }

    @Provides
    @Singleton
    Application provideApplication(){
        return mApp;
    }
}

File: Quizzical/src/main/java/com/asfour/modules/AppModule.kt
package com.asfour.modules

import com.asfour.utils.ConnectivityAssistant
import org.koin.android.ext.koin.androidContext
import org.koin.dsl.module.module

/**
 * Created by Waqqas on 15/07/15.
 */
val appModule = module {
    single { ConnectivityAssistant(androidContext()) }
}